pretyping/evarconv.ml

```diff
@@ -1820,8 +1820,14 @@ let error_cannot_unify env evd pb ?reason t1 t2 =
     ?loc:(loc_of_conv_pb evd pb) env
     evd ?reason (t1, t2)
 
-let check_problems_are_solved env evd =
-  match snd (extract_all_conv_pbs evd) with
+let check_problems_are_solved ?evars env evd =
+  let has_evar (pbty,_,t1,t2) =
+    match evars with
+    | None -> true
+    | Some evars ->
+      (try Evar.Set.mem (head_evar evd t1) evars with NoHeadEvar -> false) ||
+      (try Evar.Set.mem (head_evar evd t2) evars with NoHeadEvar -> false) in
```

Comment:
this code looks a lot like status_changed in evarsolve except the evar set is optional, maybe worth sharing